### PR TITLE
Incomplete Port grouping redo/v1

### PR DESCRIPTION
**Pull request by inashivb**

### CODE REVIEW

File: src/detect.h [hunk 1]
```diff
@@ -228,6 +229,27 @@ typedef struct DetectPort_ {
...
+int PICompare(struct PortInterval *a, struct PortInterval *b);
+
+/* red-black tree prototype for Port Groups */
+RB_HEAD(PI, PortInterval);
+RB_PROTOTYPE(PI, PortInterval, rb, PICompare);
+
+typedef struct PortIntervals_ {
+    struct PI tree;   /* red black tree of Port Intervals */
+    PortInterval *head;   /* rbtree head; should always be same as RB_MIN */
+} PortIntervals;
```
Comment (victorjulien) [inline src/detect.h:251]: Would name this `PortIntervalTree` for easier reading

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_cfg.
  

Pipeline 17512

**Comment by catenacyber** [comment]
What is expected out of this draft ?

**Comment by inashivb** [comment]
> What is expected out of this draft ?

Just a show that the work has started. This Draft is nowhere close to completion but just has the first baby step. Will add a new draft with updates soon replacing this one so this becomes a series to track this work.

**Pull request closed without merging**